Src/libCZI/utilities.h

```diff
@@ -204,3 +204,74 @@ struct ParseEnumHelper
     }
 };
 
+/// This class allows to calculate the area covered by a set of rectangles. The mode of operation
+/// is:
+/// - Create an instance of the class and add the rectangles to it (using AddRectangle) .
+/// - Then, for a given rectangle, call CalcAreaOfIntersectionWithRectangle in order to get the area of the intersection  
+///    of this rectangle with the union of the rectangles added before.
+/// The rectangles being added do not have to follow any order, or are required to be non-overlapping.
+class RectangleCoverageCalculator
+{
+private:
+    /// This vector contains the rectangles added to the state of the instance. The rectangles
+    /// in this vector are guaranteed to be non-overlapping. That is why the name 'splitters' is used -
+    /// if rectangles are which are overlapping with the existing ones, the rectangle is split smaller 
+    /// rectangles (which we call the 'splitters') which are non-overlapping with the existing ones.
+    std::vector<libCZI::IntRect> splitters_;    
+public:
+    /// Adds a rectangle to the state.
+    ///
+    /// \param  rectangle   The rectangle to be added.
+    void AddRectangle(const libCZI::IntRect& rectangle);
+
+    /// Adds the rectangles given by the iterator to the state of the instance.
+    ///
+    /// \typeparam  tIterator   Type of the iterator.
+    /// \param  begin   The begin.
+    /// \param  end     The end.
+    template <typename tIterator>
+    void AddRectangles(tIterator begin, tIterator end)
```

Comment:
We should indicate that this method's current implementation is at least quadratic in execution time with the size of the input. Might be worse under extreme circumstances (lots of splits)?

Reply:
Btw. would it make sense to try to improve on that? Or do we assume that the actual number of rectangles will always be relatively small?

Reply:
I'd have a lot of faith that the performance of this implementation is sufficient for all "remotely conceivable cases". I'd expect the number of rectangles to be in the order "a few dozen at absolute most". Until proven otherwise, I'd leave it at this.